### PR TITLE
feat: added CDN deploy bash command

### LINKED ISSUE #138: Implement a deploy procedure (1)
Implement a deploy procedure for V8 layer project.

### PR DESCRIPTION
**Pull request by hailem**
fixes #138

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+update_lit_html_references(){
+    sed -ie "s/from 'lit-html'/from '..\/lit-html\/lit-html.js'/g" $1
+    sed -ie "s/from 'lit-html/from '..\/lit-html/g" $1
+    sed -ie "s/lit-html\/directives\/unsafe-html'/lit-html\/directives\/unsafe-html.js'/g" $1
+    sed -ie "s/lit-html\/directives\/style-map'/lit-html\/directives\/style-map.js'/g" $1
+    sed -ie "s/lit-html\/directives\/repeat'/lit-html\/directives\/repeat.js'/g" $1
```
Comment (carlosescri) [inline deploy.sh]: Add all directives here… some are missing. For reference:

https://lit-html.polymer-project.org/guide/template-reference#built-in-directives

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+deploy(){
+    printf "Preparing deploy for $1 ..."
+    target_js=deploy/$1/$1.js 
+    cp -R packages/$1/lib deploy/$1
+    sed -ie "s/from 'lit-element'/from '..\/lit-element\/lit-element.js'/g" ${target_js}
+    update_lit_html_references ${target_js}
+    sed -ie "s/@dfwc\/core/..\/core\/core.js/g" ${target_js}
+    sed -ie  "s/from 'doofinder'/from '..\/doofinder\/index_md.js'/g" ${target_js}
```
Comment (carlosescri) [inline deploy.sh]: These are not the only dependencies of packages. For instance, `@dfwc/pager` has a dependency of `@dfwc/icon`. I think you should read `package.json` with the `jq` tool or something like that to get the dependencies.

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    cp -R node_modules/lit-html deploy/lit-html
+    cp -R node_modules/lit-element deploy/lit-element
+    update_lit_html_references deploy/lit-element/lit-element.js
+    cp -R node_modules/doofinder/lib deploy/doofinder
+    cp deploy/doofinder/index.mjs deploy/doofinder/index_md.js
+}
+
+# Deploy ours modules from packages
+deploy_dfcomponent_modules(){
+    modules=(close core display history icon input layer layout-selector pager panel range results sort suggestions terms total)
```
Comment (carlosescri) [inline deploy.sh]: This should read directories under `/packages` instead of providing a fixed list of modules.

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    for module in ${modules[@]}; do
+        deploy ${module[@]}
+    done
+}
+
+# Upload files to S3
+deploy_S3_files(){
+    printf "Uploading files to S3 ... \n"
+    aws s3 sync ./deploy/ s3://doofinder-cdn/dev/v8 --delete --acl public-read
```
Comment (carlosescri) [inline deploy.sh]: Did you try the `--metadata` flag?

```
--metadata '{"Cache-Tag":"webcomponents"}'
```

File: deploy.sh (hunk 1 above)
Comment (hailem) [reply deploy.sh]: ok

File: deploy.sh (hunk 2 above)
Comment (hailem) [reply deploy.sh]: all @dfwc/<anything> is converter to ../anything/anything.js

File: deploy.sh (hunk 3 above)
Comment (hailem) [reply deploy.sh]: done

File: deploy.sh (hunk 4 above)
Comment (hailem) [reply deploy.sh]: I have tried with etag using --metadata

File: deploy.sh [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    mkdir deploy
+}
+
+# Deploy lit-element, lit-html and doofinder from node_modules
+deploy_external_modules(){
+    cp -R node_modules/lit-html deploy/lit-html
+    cp -R node_modules/lit-element deploy/lit-element
+    update_lit_html_references deploy/lit-element/lit-element.js
+    cp -R node_modules/doofinder/lib deploy/doofinder
+    cp deploy/doofinder/index.mjs deploy/doofinder/index_md.js
```
Comment (carlosescri) [inline deploy.sh]: Why not…?

```
mkdir deploy/doofinder
cp node_modules/doofinder/lib/index.mjs deploy/doofinder/doofinder.js
```

**Pull request merged**